src/library/librarycontrol.cpp

```diff
@@ -405,16 +405,25 @@ void LibraryControl::emitKeyEvent(QKeyEvent&& event) {
 }
 
 void LibraryControl::setLibraryFocus() {
-    if (m_pSidebarWidget) {
-        // XXX: Set the focus of the library panel directly instead of sending tab from sidebar
-        m_pSidebarWidget->setFocus();
+    // XXX: Set the focus of the library panel directly instead of sending tab from sidebar
+    VERIFY_OR_DEBUG_ASSERT(m_pSidebarWidget) {
+        return;
+    }
+    // Try to focus the sidebar.
+    m_pSidebarWidget->setFocus();
+    // This may have failed, for example when a Cover window has focus,
+    // so make sure the sidebar is focused or we'll crash.
+    if (m_pSidebarWidget->hasFocus()) {
```

Comment:
I would suggest using an early return here. Then we can split the following two sentences of comments, one for each possible outcome.

Reply:
I'd go with this:
```c++
    // Try to focus the sidebar.
    m_pSidebarWidget->setFocus();
    // This may have failed, for example when a Cover window still has focus,
    // so make sure the sidebar is focused or we'll crash.
    if (!m_pSidebarWidget->hasFocus()) {
        return;
    }
    // Send Tab to move focus to the Tracks table.
    // Obviously only works as desired if the skin widgets are arranged
    // accordingly.
    QKeyEvent event(QEvent::KeyPress, Qt::Key_Tab, Qt::NoModifier);
    QApplication::sendEvent(m_pSidebarWidget, &event);
```
?